standards/secrets-management.md

```diff
@@ -0,0 +1,30 @@
+---
+expires: 2019-03-15
+---
+# Secrets management
+
+'Secrets' is defined here as a sensitive bit of information that should be kept private but with a technical system or user focus, such as, a password, OAuth token or SSH private key.
+
+'Secrets' here is **not** referring to the `SECRET` classification.
+
+## The base principle
+
+All secrets **must** be adequately protected from a loss of confidentiality or integrity. Secrets, much like other confidential data, must only be accessible or otherwise open to influence from intended authorised parties.
+
+## Application & infrastructure secrets
+
+All secrets should be adequately protected and suitably stored.
+
+Where possible, use infrastructure-based secrets management services such as [AWS Key Management Service](https://aws.amazon.com/kms/), [AWS Systems Manager Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-paramstore.html), [Microsoft Azure Key Vault](https://azure.microsoft.com/en-gb/services/key-vault/) or [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) on MOJ's Cloud Platforms.
+
+It should be rare and exceptional to store secrets within code repositories, such as in Github.com, and where conducted must use [git-crypt](https://github.com/AGWA/git-crypt) to encrypt those secrets and control who has the ability to view (decrypt) those secrets.
+
+Secrets must never be stored in plain-text, particularly in code repositories.
```

Comment:
I think it could be useful to show some examples here of things people might need to do and how we'd prefer them to do it.

Reply:
Also, what about environment injection of secrets via envars?